rust-runtime/aws-smithy-runtime-api/src/client/http.rs

```diff
@@ -0,0 +1,305 @@
+//! Client, and then configure some override connect/read timeouts for a specific operation. These timeouts
+//! ultimately are part of the connector, so the same connector can't be reused for the two different sets
+//! of timeouts. Thus, the [`HttpClient`] implementation is responsible for managing multiple connectors
+//! with varying config. Some example configs that can impact which connector is used:
+//!
+//! - HTTP protocol versions
+//! - TLS settings
+//! - Timeouts
+//!
+//! Some of these aren't implemented yet, but they will appear in the [`HttpConnectorSettings`] struct
+//! once they are.
+
+use crate::client::orchestrator::{HttpRequest, HttpResponse};
+use crate::client::runtime_components::RuntimeComponents;
+use crate::impl_shared_conversions;
+use aws_smithy_async::future::now_or_later::NowOrLater;
+use aws_smithy_http::result::ConnectorError;
+use pin_project_lite::pin_project;
+use std::fmt;
+use std::future::Future as StdFuture;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::Poll;
+use std::time::Duration;
+
+type BoxFuture = Pin<Box<dyn StdFuture<Output = Result<HttpResponse, ConnectorError>> + Send>>;
+
+pin_project! {
+    /// Future for [`HttpConnector::call`].
+    pub struct HttpConnectorFuture {
+        #[pin]
+        inner: NowOrLater<Result<HttpResponse, ConnectorError>, BoxFuture>,
+    }
+}
+
+impl HttpConnectorFuture {
+    /// Create a new `HttpConnectorFuture` with the given future.
+    pub fn new<F>(future: F) -> Self
+    where
+        F: StdFuture<Output = Result<HttpResponse, ConnectorError>> + Send + 'static,
+    {
+        Self {
+            inner: NowOrLater::new(Box::pin(future)),
+        }
+    }
+
+    /// Create a new `HttpConnectorFuture` with the given boxed future.
+    ///
+    /// Use this if you already have a boxed future to avoid double boxing it.
+    pub fn new_boxed(
+        future: Pin<Box<dyn StdFuture<Output = Result<HttpResponse, ConnectorError>> + Send>>,
+    ) -> Self {
+        Self {
+            inner: NowOrLater::new(future),
+        }
+    }
+
+    /// Create a `HttpConnectorFuture` that is immediately ready with the given result.
+    pub fn ready(result: Result<HttpResponse, ConnectorError>) -> Self {
+        Self {
+            inner: NowOrLater::ready(result),
+        }
+    }
+}
+
+impl StdFuture for HttpConnectorFuture {
+    type Output = Result<HttpResponse, ConnectorError>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut std::task::Context<'_>) -> Poll<Self::Output> {
+        let this = self.project();
+        this.inner.poll(cx)
+    }
+}
+
+/// Trait with a `call` function that asynchronously converts a request into a response.
+///
+/// Ordinarily, a connector would use an underlying HTTP library such as [hyper](https://crates.io/crates/hyper),
+/// and any associated HTTPS implementation alongside it to service requests.
+///
+/// However, it can also be useful to create fake connectors implementing this trait
```

Comment:
```suggestion
/// However, it can also be useful to create mock connectors implementing this trait
```
I think using `mock` would help lead customers do a text search to find the test connectors.